src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -413,7 +413,23 @@ private static int getTokenType(ParseTree node) {
         final int tokenType;
 
         if (node.getChildCount() == 0) {
-            tokenType = ((TerminalNode) node).getSymbol().getType();
+            final int antlrTokenType = ((TerminalNode) node).getSymbol().getType();
+            // Newly added HTML tag names' tokens have to be introduced before the HTML_TAG_NAME
+            // token in JavadocLexer.g4. As a result the value that ANTLR autromatically
+            // generates for HTML_TAG_NAME token type changes. On the other hand checkstyle needs
+            // to provide consistent token values to the users in JavadocTokenTypes.java to avoid
+            // loss of compatability with the older versions. Accordingly the token value for
+            // HTML_TAG_NAME and one other token (EMBED_HTML_TAG_NAME) are hardcoded and swapped in
+            // JavadocTokenTypes.java. (See Pull #5455)
+            if (antlrTokenType == JavadocTokenTypes.HTML_TAG_NAME) {
```

Comment:
Last question then,
> token is the token object form ANTLR
> like since JavadocTokenTypes.HTML_TAG_NAME == JavadocParser.HTML_TAG_NAME we were previously able to use JavadocTokenTypes.HTML_TAG_NAME, but now since those two values have changed, we are forced to use value directly from the JavadocParser.

What if we move this to a single method for both places so this is clearer when we are dealing with antlr tokens and when we are dealing with token types?
Example:
`````
getAntlrTokenType(int antlrTokenType) {
final int tokenType;
            if (antlrTokenType == JavadocTokenTypes.HTML_TAG_NAME) {
                tokenType = JavadocTokenTypes.EMBED_HTML_TAG_NAME;
            }
            else if (antlrTokenType == JavadocTokenTypes.EMBED_HTML_TAG_NAME) {
                tokenType = JavadocTokenTypes.HTML_TAG_NAME;
            }
            else {
                tokenType = antlrTokenType;
            }

return tokenType;
}
`````
This would prevent us from duplicating the same conversion in 2 places, and if we decide to move forward with a map conversion, it will be a simplier replacement.

Note: instead of `int antlrTokenType`, it also could be `Token token` because `((TerminalNode) node).getSymbol()` returns `Token` and `token` in the other section is already of type `Token`.

Reply:
Sorry I am unable to get what are you trying to refer to when you say, "duplicating the same conversion in 2 places".

Do you intend to call "getAntlrTokenType" in place of token.getType() at [JavadocDetailNodeParser.java#L505](https://github.com/ps-sp/checkstyle/blob/51cf1ced9409cb63eeb01fade4703422017d54a4/src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java#L505) so that we can use JavadocTokenTypes.HTML_TAG_NAME instead of JavadocParser.HTML_TAG_NAME ? May be I am wrong but I can't see any need for such a method.

Reply:
> getAntlrTokenType

Name is just an exmaple but maybe `getTokenType` or `getJavadocTokenType` would make more sense.

> Do you intend to call "getAntlrTokenType" in place of token.getType() at JavadocDetailNodeParser.java#L505 so that we can use JavadocTokenTypes.HTML_TAG_NAME instead of JavadocParser.HTML_TAG_NAME 

Yes, 505 and 424 would be replaced with these calls.

> May be I am wrong but I can't see any need for such a method.

To simplify the process of converting antlr token to token type and to maintain a consistency  we only use token types in all our code. To me it is not clear, when we are dealing with one versus the other and I thought it would make more sense we stick to using token types, since this what will be found in all our checks that use the Javadoc parse. We then isolate all `JavadocParser` uses to a single method.

So 424 would become: `tokenType = getAntlrTokenType(((TerminalNode) node).getSymbol());`
and 505 would be `final int type = getAntlrTokenType(tokenList.get(i));` and every if statement following it would continue to use `JavadocTokenTypes`.

If we keep things as they are and don't make this method, shouldn't `START` at 506 be `JavadocParser` and not `JavadocTokenTypes`?

@romani Please share your thoughts.

Reply:
@rnveach Yeah, I agree with the consistency points you made. Okay I will update PR.